cypress/e2e/tests/migration/applicationinventory/applications/interlinked.test.ts

```diff
@@ -155,6 +156,79 @@ describe(["@tier3"], "Applications interlinked to tags and business service", ()
         cy.get(continueButton).click();
     });
 
+    it("Validates association application tags to  archetype tags ", function () {
+        //automates polarion MTA-401
+        //create archytype
+        tagList = createMultipleTags(3);
+        const archetype = new Archetype(
+            data.getRandomWord(8),
+            [tagList[0].name, tagList[2].name],
+            [tagList[1].name],
+            null
+        );
+        archetype.create();
+        cy.wait(2 * SEC);
+
+        //Create 3 appdatas containing different tags
+        let appdata1 = {
+            name: data.getAppName(),
+            description: data.getDescription(),
+            tags: [tagList[0].name, tagList[2].name],
+            comment: data.getDescription(),
+        };
+        let appdata2 = {
+            name: data.getAppName(),
+            description: data.getDescription(),
+            tags: [tagList[0].name, tagList[1].name],
+            comment: data.getDescription(),
+        };
+        let appdata3 = {
+            name: data.getAppName(),
+            description: data.getDescription(),
+            tags: [tagList[0].name, tagList[2].name],
+            comment: data.getDescription(),
+        };
+
+        //assign different appdata to each app and check if tags are present
+        const application1 = new Application(appdata1);
+        applicationList.push(application1);
+        application1.create();
+        cy.get("@getApplication");
+        cy.wait(2 * SEC);
+        application1.applicationDetailsTab("Tags");
+        appdata1.tags.forEach((tag) => {
+            application1.tagAndCategoryExists(tag);
+        });
+        application1.closeApplicationDetails();
+
+        const application2 = new Application(appdata2);
+        applicationList.push(application2);
+        application2.create();
+        cy.get("@getApplication");
+        cy.wait(2 * SEC);
+        application2.applicationDetailsTab("Tags");
+        appdata2.tags.forEach((tag) => {
+            application2.tagAndCategoryExists(tag);
+        });
+        application2.closeApplicationDetails();
+
+        const application3 = new Application(appdata3);
+        applicationList.push(application3);
+        application3.create();
+        cy.get("@getApplication");
+        cy.wait(2 * SEC);
+        application3.applicationDetailsTab("Tags");
+        appdata3.tags.forEach((tag) => {
+            application3.tagAndCategoryExists(tag);
+        });
+        application3.closeApplicationDetails();
```

Comment:
Same here, you can iterate within array, don't duplicate yourself

Reply:
Done